src/neo4clj/query_builder.clj

```diff
@@ -8,19 +8,31 @@
   [{:keys [ref-id] :as node} return?]
   (str "CREATE " (cypher/node node) (when return? (str " RETURN " ref-id))))
 
-(defn lookup-query
-  "Takes a lookup representation and generates a bolt query
-
-  A lookup representation needs the :reference.id to be set and
-  either the :id or :labels and :properties keys"
-  [{:keys [ref-id] :as lookup} return?]
-  (let [cypher-lookup (cypher/lookup lookup)]
+(defn lookup
+  [entity-fn {:keys [ref-id] :as entity} return?]
+  (let [[base-cypher where-cypher] (entity-fn entity)]
     (str "MATCH "
-         (str (first cypher-lookup)
-              (when (second cypher-lookup)
-                (str " WHERE " (second cypher-lookup))))
+         (str base-cypher
+              (when where-cypher
+                (str " WHERE " where-cypher)))
          (when return? (str " RETURN " ref-id)))))
```

Comment:
I think you can remove a `str` (nesting):
```clojure
     ...
     (str "MATCH "
         base-cypher
         (when where-cypher
           (str " WHERE " where-cypher))
         (when return?
           (str " RETURN " ref-id)))))
```

Reply:
Fixed